sdk/servicebus/azure-messaging-servicebus/src/samples/java/com/azure/messaging/servicebus/receivesettle/async/MessageReceiverAsyncWorker.java

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.servicebus.receivesettle.async;
+
+import com.azure.core.util.IterableStream;
+import com.azure.messaging.servicebus.ServiceBusReceivedMessage;
+import com.azure.messaging.servicebus.ServiceBusReceivedMessageContext;
+import com.azure.messaging.servicebus.ServiceBusReceiverAsyncClient;
+import com.azure.messaging.servicebus.receivesettle.NetworkFailureException;
+import com.azure.messaging.servicebus.receivesettle.Order;
+import com.azure.messaging.servicebus.receivesettle.OrderServiceFailureException;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * MessageReceiverAsyncApp receives messages then use one or multiple MessageReceiverAsyncWorker
+ * to process the messages.
+ * This sample code uses only one MessageReceiverAsyncWorker.
+ */
+public class MessageReceiverAsyncWorker {
+    private final ServiceBusReceiverAsyncClient receiverClient;
+    private final OrderAsyncService orderService;
+
+    public MessageReceiverAsyncWorker(ServiceBusReceiverAsyncClient receiverClient, OrderAsyncService orderService) {
+        this.receiverClient = receiverClient;
+        this.orderService = orderService;
+    }
+
+    /**
+     * Convert a message to order and call the order service to process it.
+     * Handle exceptions thrown from the order service.
+     *
+     * @param messageContext The message context that includes the received message.
+     * @return A Mono.
+     */
+    public Mono<Void> processMessageOneByOne(ServiceBusReceivedMessageContext messageContext) {
+        ServiceBusReceivedMessage message = messageContext.getMessage();
+        Order order = convertMessageToOrder(message);
+        return orderService.createOrReplaceOrder(order)
```

Comment:
If createOrReplaceOrder is a time-consuming operation, we should also demonstrate how to renew locks on the message.

Reply:
If a user's service doesn't run longer time than the initial lock very often and the service is idempotent, this auto renew isn't required I think. Anyway, I added `.maxAutoLockRenewalDuration(Duration.ofSeconds(10))` to have auto renew for demo purpose.